logical/framework/path_map_test.go

```diff
@@ -444,3 +329,20 @@ func TestPathMap_SaltFunc(t *testing.T) {
 		t.Fatal("did not find upgraded value")
 	}
 }
+
+func TestPathMap_SaltFunc(t *testing.T) {
+	storage := new(logical.InmemStorage)
+
+	salt, err := saltpkg.NewSalt(storage, &saltpkg.Config{
+		HashFunc: saltpkg.SHA1Hash,
+	})
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	saltFunc := func() (*saltpkg.Salt, error) {
+		return salt, nil
+	}
+
+	testSalting(t, storage, salt, &PathMap{Name: "foo", SaltFunc: saltFunc})
+}
```

Comment:
I would really love to also see the original error report reproduced via TestCluster in a user-API-oriented test (it's not difficult) and verify that it fails with the original code, then verify it passes here.

Reply:
Done.